packages/flutter/lib/src/cupertino/text_field.dart

```diff
@@ -773,20 +811,25 @@ class _CupertinoTextFieldState extends State<CupertinoTextField> with AutomaticK
                 : CupertinoTheme.of(context).brightness == Brightness.light
                     ? _kDisabledBackground
                     : CupertinoColors.darkBackgroundGray,
-            child: TextSelectionGestureDetector(
-              onTapDown: _handleTapDown,
-              onForcePressStart: _handleForcePressStarted,
-              onForcePressEnd: _handleForcePressEnded,
-              onSingleTapUp: _handleSingleTapUp,
-              onSingleLongTapStart: _handleSingleLongTapStart,
-              onSingleLongTapMoveUpdate: _handleSingleLongTapMoveUpdate,
-              onSingleLongTapEnd: _handleSingleLongTapEnd,
-              onDoubleTapDown: _handleDoubleTapDown,
-              onDragSelectionStart: _handleMouseDragSelectionStart,
-              onDragSelectionUpdate: _handleMouseDragSelectionUpdate,
-              onDragSelectionEnd: _handleMouseDragSelectionEnd,
-              behavior: HitTestBehavior.translucent,
-              child: _addTextDependentAttachments(paddedEditable, textStyle, placeholderStyle),
+            child: Listener(
+              onPointerDown: (PointerDownEvent event) {
+                _lastUsedDeviceKind = event.kind;
```

Comment:
You should decide on this, though while we want to really keep this simple, this may be a bit too simple :D. I think you might still want a little bit of the pointer tracking here. Unless you think the pointer events will never be in a situation where the kinds will mix.

Reply:
This is by design. We only want to take into account the actions that "trigger" the text field. All actions that "trigger" the text field start with a PointerDown event (tap, long press, drag, long-press-drag, etc).

See the [internal Q/A](https://yaqs.googleplex.com/eng/q/4519774183489536) for the opinion of the Material team on this.

Reply:
This is really an edge case, but I was asking what is supposed to happen if you held down and dragged a finger on Android, selecting some words, and right before you lift the finger, there was also a mouse click somewhere. 

By your rule above, the 'trigger' is the touch, therefore it should show the toolbar, but by the current implementation, the last kind is mouse and there won't be a toolbar.